File: 2087/Conjuntos Bons e Ruins.py
n = int(input())


while n!=0:
    wordsList = []
    flag = False
    for i in range(n):
        word = input()
        if not flag:
            for words in wordsList:
                if word.startswith(words) or words.startswith(word):
                    #print(word)
                    flag = True
                    break
        
        wordsList.append(word)

    if flag:
        print("Conjunto Ruim")
    else:
        print("Conjunto Bom")
    
    flag = False
    wordsList.clear()
    n = int(input())

File: 2087/Conjuntos Bons e Ruins.cpp
#include <bits/stdc++.h>
#include <string>
#include <iostream>
using namespace std;

bool startsWith(std::string mainStr, std::string toMatch)
{
    // std::string::find returns 0 if toMatch is found at starting
    if(mainStr.find(toMatch) == 0)
        return true;
    else
        return false;
}

int main (){

    int n; 
    cin >> n;
    vector<string> wordsList;
    bool flag = false;
    
    
    while (n!=0){
        
        for(int i = 0; i < n; i++){
            string word;
            cin >> word;

            if (!flag){
                for (int j = 0; j < wordsList.size(); j++){
                    if (startsWith(word, wordsList[j]) || startsWith(wordsList[j], word)){
                        flag = true;
                        break;
                    }
                }
            }

            wordsList.push_back(word);
        }
        if (flag) cout << "Conjunto Ruim\n";
        else cout << "Conjunto Bom\n";

        flag = false;
        wordsList.clear();
        cin >> n;
       
        

        
        
    }
}
